spectral_cube/base_class.py

```diff
@@ -554,30 +561,53 @@ def identify_bad_beams(self, threshold, reference_beam=None,
         return includemask
 
     def average_beams(self, threshold, mask='compute', warn=False):
+        '''
+        This is now deprecated. See `~VaryingResolutionSpectralCube.set_common_beam.`.
+        '''
+
+        warnings.warn("average_beams is deprecated and its functionality has been removed. "
+                      "This is because beam averaging is not the correct operation. Please use"
+                      " `cube.set_common_beam` in its place to compute the common beam. The "
+                      "common beam can then be accessed with `cube.common_beam`.",
+                      DeprecationWarning)
+
+        self.set_common_beam(threshold, mask=mask, warn=warn)
+
+    def set_common_beam(self,
+                        threshold=None, mask='compute', warn=False,
+                        combeam_kwargs={}):
```

Comment:
I'm not a fan of the ``combeam`` abbreviation - maybe just ``common_beam_kwargs``, or why not simply have ``**kwargs`` and pass them through to the common beam algorithm? Is there a risk of overlap? Since the main purpose of the method is to actually call the common beam algorithm, just using ``**kwargs`` would seem more natural to me?

Reply:
Agreed. I've switched to `**kwargs` here.